relay-server/src/actors/store.rs

```
@@ -762,6 +780,16 @@ impl Handler<StoreEnvelope> for StoreForwarder {
                     start_time,
                     item,
                 )?,
+                ItemType::ReplayPayload => {
+                    debug_assert!(topic == KafkaTopic::ReplayPayloads);
+                    let replay_payload = self.produce_attachment_chunks(
+                        event_id.ok_or(StoreError::NoEventId)?,
+                        scoping.project_id,
+                        item,
+                        topic,
+                    )?;
+                    replay_payloads.push(replay_payload);
```

Instead of collecting the replay payloads into a vector and producing them at the end, produce them directly as you're iterating. 

With the current logic, you would silently drop replay payloads if there happens to be an event in the same envelope (which should be supported from SDK side imho).